components/profile-controller/controllers/profile_controller.go

```diff
@@ -443,11 +448,22 @@ func (r *ProfileReconciler) getAuthorizationPolicy(profileIns *profilev1.Profile
 				From: []*istioSecurity.Rule_From{{
 					Source: &istioSecurity.Source{
 						Principals: []string{
-							"cluster.local/ns/istio-system/sa/istio-ingressgateway-service-account",
+							istioIGWPrincipal,
 						},
 					},
 				}},
 			},
+			{
+				From: []*istioSecurity.Rule_From{
+					{
+						// KFP UI needs to talk to ml-pipeline-ui-artifact pods
+						// in each profile namespace
+						Source: &istioSecurity.Source{
+							Principals: []string{kfpUIPrincipal},
+						},
+					},
+				},
+			},
```

Comment:
This rule should also include the "when" condition, to prevent the pipeline UI from being able to proxy traffic without a userid-header.

## Option 1

The best option is to simply add another "from" principal  in the first rule, and remove this rule.

```go
			{
				When: []*istioSecurity.Condition{
					{
						Key: fmt.Sprintf("request.headers[%v]", r.UserIdHeader),
						Values: []string{
							r.UserIdPrefix + profileIns.Spec.Owner.Name,
						},
					},
				},
				From: []*istioSecurity.Rule_From{{
					Source: &istioSecurity.Source{
						Principals: []string{
							// end-users access via the ingress gateway
							istioIGWPrincipal,
							// kfp proxies user requests to artifact pods in profile namespaces
							kfpUIPrincipal,
						},
					},
				}},
			},
```

## Option 2

We could keep it as a separate rule, and add the "when" condition, but this is is less efficient for Envoy, and needless because it has the same effect.

```go
			{
				// kfp proxies user requests to artifact pods in profile namespaces
				When: []*istioSecurity.Condition{
					{
						Key: fmt.Sprintf("request.headers[%v]", r.UserIdHeader),
						Values: []string{
							r.UserIdPrefix + profileIns.Spec.Owner.Name,
						},
					},
				},
				From: []*istioSecurity.Rule_From{
					{
						Source: &istioSecurity.Source{
							Principals: []string{kfpUIPrincipal},
						},
					},
				},
			},
```